### ISSUE TITLE
Allow for arbitrary input FASTQ filenames to be used

### ISSUE THREAD

**Issue by dariober** [elsayed-lab/utr_analysis#2]
Hello,
We are trying to execute utr_analysis.py and it seems input file have to be named in a particular way to be recognized by the program.
Would it be possible for you to provide some examples of input file names that are suitable?

Many thanks!
Dario

**Comment by khughitt** [comment]
Hi Dario,

Thanks for bringing this to my attention. The pipeline is indeed currently limited to the filenames for input FASTQ files that it can handle -- this is something that I've been meaning to address for a while, but have not come up with any elegant solution so far.

Currently, the regular expression used to detect valid input files is:

`^(.*/)?([A-Z0-9]*)_(.*)(R[1-2])_(.+)\.fastq(\.gz)?`

This means:

1) [Optional] Any combination of directories (e.g. "/path/to/input/files/")
2) [Optional] Any combination of uppercase letters and numbers ("SAMPLE001")
3) An underscore
4) "R1" or "R2"
5) An underscore
6) One or more other characters (e.g. "_filtered_etc")
7) .fastq   (all lowercase)
8) [Optional] .gz

So a valid example, would be:

```
"/path/to/input/data/6973_R1_filtered_adaptertrim.fastq.gz"
```

The corresponding argument to `utr_analysis.py` might then be:

```
utr_analysis.py -i "/path/to/input/data/*.fastq.gz"
```

Note that the quotes above are important for Python to properly handle the wildcard expressions.

That said, the above requirements are very arbitrary and are simply based on the conventions used by the files I've encountered so far.

I'll try and update the pipeline to relax the filename requirements and do a better job describing any requirements in the README.

In the meantime though, if you want to start running the pipeline, you should be able to modify your filenames to follow the above conventions and get it working.

Let me know if you run into any further issues. Patches are of course welcome if you would like to take the initiative to improve the pipeline yourself.

Best,
Keith

**Comment by khughitt** [comment]
Update: Samples are also required to be separated into sub-directores, e.g.

.../sample01/file_R1.fastq
                    file_R2.fastq
.../sample02/file_R1.fastq
                    file_R2.fastq

Going to remove that requirement as well.

**Comment by khughitt** [comment]
Hi Dario,

I made a couple more updates to the pipeline code over the weekend and it should no longer be quite so picky with the directory structure and filenames used for input reads.

Test it out and let me know if you still have trouble.

In case you find it useful, I also posted some knitr/R code for analyzing the output from this pipeline: https://github.com/elsayed-lab/utr_length_composition_and_site_usage_analysis

Feel free to use any or all of the code that you find interesting.

If you have any ideas or suggestions, you are welcome to file further issues, or submit pull requests with improvements directly.

All the best,
Keith
